corehq/apps/hqadmin/views.py

```diff
@@ -225,6 +229,46 @@ def contact_email(request):
     return response
 
 
+class AuthenticateAs(BasePageView):
+    urlname = 'authenticate_as'
+    page_title = _("Login as other user")
+    template_name = 'hqadmin/authenticate_as.html'
+
+    @property
+    @memoized
+    def authenticate_as_form(self):
+        if self.request.method == 'POST':
+            return AuthenticateAsForm(self.request.POST)
+        else:
+            return AuthenticateAsForm()
+
+    @method_decorator(require_superuser)
+    def dispatch(self, *args, **kwargs):
+        return super(AuthenticateAs, self).dispatch(*args, **kwargs)
+
+    def page_url(self):
+        return reverse(self.urlname)
+
+    @property
+    def page_context(self):
+        return {
+            'hide_filters': True,
+            'form': self.authenticate_as_form
+        }
+
+    def post(self, request, *args, **kwargs):
```

Comment:
i think the dispatch decorator covers this, but just confirming this doesn't also need one